src/test/java/com/glencoesoftware/omero/ms/image/region/HistogramRequestHandlerTest.java

```diff
@@ -0,0 +1,100 @@
+
+        params.add("imageId", "1");
+        params.add("theC", "0");
+        params.add("z", "0");
+        params.add("t", "0");
+        params.add("maxPlaneWidth", "2048");
+        params.add("maxPlaneHeight", "2048");
+
+        histogramCtx = new HistogramCtx(params, "");
+        requestHandler = new HistogramRequestHandler(histogramCtx, null);
+    }
+
+    @Test
+    public void testHistogramData() {
+        ByteBuffer bb = ByteBuffer.allocate(4);
+        bb.put(new byte[] {1,2,3,4});
+        PixelData pd = new PixelData(PixelsType.VALUE_UINT8, bb);
+        int imgWidth = 256;
+        int imgHeight = 512;
+        double[] minMax = new double[] {0, 4};
+        histogramCtx.bins = 4;
+        JsonArray testData = requestHandler.getHistogramData(pd,
+                minMax, imgWidth, imgHeight);
+        Assert.assertEquals(testData.size(), 4);
+        Assert.assertEquals(testData.getValue(0), 1);
+        Assert.assertEquals(testData.getValue(1), 1);
+        Assert.assertEquals(testData.getValue(2), 1);
+        Assert.assertEquals(testData.getValue(3), 1);
+    }
+
+    @Test(expected=IllegalArgumentException.class)
+    public void testValGreaterThanMax() {
+        ByteBuffer bb = ByteBuffer.allocate(4);
+        bb.put(new byte[] {1,2,3,4});
+        PixelData pd = new PixelData(PixelsType.VALUE_UINT8, bb);
+        int imgWidth = 256;
+        int imgHeight = 512;
+        double[] minMax = new double[] {0, 3};
+        histogramCtx.bins = 4;
+        JsonArray testData = requestHandler.getHistogramData(pd,
+                minMax, imgWidth, imgHeight);
+    }
+
+    @Test(expected=IllegalArgumentException.class)
+    public void testValLessThanMin() {
+        ByteBuffer bb = ByteBuffer.allocate(4);
+        bb.put(new byte[] {1,2,3,4});
+        PixelData pd = new PixelData(PixelsType.VALUE_UINT8, bb);
+        int imgWidth = 256;
+        int imgHeight = 512;
+        double[] minMax = new double[] {2, 4};
+        histogramCtx.bins = 4;
+        JsonArray testData = requestHandler.getHistogramData(pd,
+                minMax, imgWidth, imgHeight);
+    }
+
+    @Test
+    public void testMoreBinsThanValues() {
+        ByteBuffer bb = ByteBuffer.allocate(3);
+        bb.put(new byte[] {0,1,3});
+        PixelData pd = new PixelData(PixelsType.VALUE_UINT8, bb);
+        int imgWidth = 256;
+        int imgHeight = 512;
+        double[] minMax = new double[] {0, 3};
+        histogramCtx.bins = 5;
+        JsonArray testData = requestHandler.getHistogramData(pd,
+                minMax, imgWidth, imgHeight);
+        Assert.assertEquals(5, testData.size());
+        Assert.assertEquals(1, (int) testData.getInteger(0));
+        Assert.assertEquals(0, (int) testData.getInteger(1));
+        Assert.assertEquals(1, (int) testData.getInteger(2));
+        Assert.assertEquals(0, (int) testData.getInteger(3));
+        Assert.assertEquals(1, (int) testData.getInteger(4));
+    }
```

Comment:
In addition to the tests already implemented, a couple of other scenarios that might be worth adding to the unit tests:

- a min/max range wider that the dynamic range e.g. a 0-10 range for real values 4,5,6
- floating point values assuming floating-point data is in scope for this endpoint
- a pixel data array larger than `imgWidth * imageWeight`